### ISSUE TITLE
Yaml-Cpp fails to build on Windows | MSVC | Premake

### ISSUE THREAD

**Issue by harshit2608** [jbeder/yaml-cpp#1137]
Specifications:
- Project Generator: Premake v5.0.0-beta2
- OS: Windows 11
- Compiler: MSVC
- Yaml-Cpp Version: Latest Master

Error:
![image](https://user-images.githubusercontent.com/42277796/193423351-6283ce53-8ce0-44b5-9c4c-f60e3890bc23.png)

Error Code: 
![image](https://user-images.githubusercontent.com/42277796/193423386-b5f67353-5036-44fa-907a-69eac6c8754f.png)
File - node_data.cpp

**Comment by jbeder** [comment]
Can you paste the exact command line to generate the MSVC project?

**Comment by nepp95** [comment]
I have this exact problem as well, also with premake. Probably the command is wrongly generated. I don't have the exact command since VS just tells us which file is being compiled instead of the whole command. I do however have the command line flags from VS studio:

`/JMC /MP /ifcOutput "..\..\..\bin-int\Debug-windows-x86_64\yaml-cpp\" /GS /W3 /Zc:wchar_t /I"include" /ZI /Gm- /Od /Fd"..\..\..\bin\Debug-windows-x86_64\yaml-cpp\yaml-cpp.pdb" /Zc:inline /fp:precise /D "_UNICODE" /D "UNICODE" /errorReport:prompt /WX- /Zc:forScope /RTC1 /Gd /MDd /std:c++17 /FC /Fa"..\..\..\bin-int\Debug-windows-x86_64\yaml-cpp\" /EHsc /nologo /Fo"..\..\..\bin-int\Debug-windows-x86_64\yaml-cpp\" /Fp"..\..\..\bin-int\Debug-windows-x86_64\yaml-cpp\yaml-cpp.pch" /diagnostics:column `

Short summary: 
- Static library
- cpp17
- staticruntime OFF (On also gives this error)
- files to compile: src/**.h, src/**.cpp, include/**.h
- includedirs: include

Project file: (Renamed to .txt to make it uploadable)
[yaml-cpp.txt](https://github.com/jbeder/yaml-cpp/files/9741518/yaml-cpp.txt)

If you need anything more, please tell me and I will provide :)

**Comment by jbeder** [comment]
Can you provide the command used to generate (with CMake) the project file?

**Comment by nepp95** [comment]
> Can you provide the command used to generate (with CMake) the project file?

CMake is not used to generate the project file. [Premake ](https://premake.github.io/)is a generator itself. It's configuration is:

```
project "yaml-cpp"
	kind "StaticLib"
	language "C++"
	cppdialect "C++17"
	staticruntime "on"

	targetdir ("%{wks.location}/bin/" .. outputdir .. "/%{prj.name}")
	objdir ("%{wks.location}/bin-int/" .. outputdir .. "/%{prj.name}")

	files {
		"src/**.h",
		"src/**.cpp",
		"include/**.h"
	}

	includedirs {
		"include"
	}

	filter "system:windows"
		systemversion "latest"

	filter "configurations:Debug"
		runtime "Debug"
		symbols "on"

	filter "configurations:Release"
		runtime "Release"
		optimize "on"
```

**Comment by jbeder** [comment]
Oh, I see. I'm guessing this is a problem with the premake configuration, then. I'll close this issue, and please ask on Stack Overflow. If it turns out to be a yaml-cpp problem, please reopen the issue with a more specific problem.

**Comment by nepp95** [comment]
I managed to fix it by adding a compiler definition. Looking through the cmake file, I saw that "YAML_CPP_STATIC_DEFINE" gets defined if it is not a shared lib. Adding that to the premake configuration fixed the problem.
Weirdly enough it only gave an error on 1 variable, but hey, it works :)

```
.......

defines {
	"YAML_CPP_STATIC_DEFINE"
}

.......
```

@jbeder
I can create a pull request if you want to support Premake. It's just 1 lua file in the root dir for configuration.
